drivers/overlay/ov_network.go

```diff
@@ -46,10 +47,15 @@ type network struct {
 	once      *sync.Once
 	initEpoch int
 	initErr   error
-	subnets   []*subnet
+	*networkConfiguration
 	sync.Mutex
 }
 
+type networkConfiguration struct {
+	subnets          []*subnet
+	disableDefaultGW bool
+}
```

Comment:
Based on my previous comment, most of these changes will go away. Hence I will stop my review here and wait for your updated diffs.